### PR TITLE
feat(wash-lib, wash-cli): support overriding manifest via project config

### PR DESCRIPTION
**Pull request by vados-cosmonic**

### CODE REVIEW

**Comment by vados-cosmonic** [comment]
Blocked on https://github.com/wasmCloud/wasmCloud/pull/3209  and https://github.com/wasmCloud/wasmCloud/pull/3198

**Comment by vados-cosmonic** [comment]
Note for reviewers -- this functionality includes some changes to what we thought the files would look like. 

This PR intentionally only covers the *first* suite of overrides (top level manifest changes), but the `wasmcloud.toml` edit area looks something like this now (see the added test!):

```toml
[dev]
manifests = [
  { component_name = "http-handler", path = "test.wadm.yaml" }
]
```

Where originally the RFC looked like this:

```toml
[dev]
manifest = "wadm.yaml"
```

There are some reasons for this:

- The deployment of a project could consist of *multiple* manifests (and certainly will once we're doing workspace-support) -- we probably do not want to limit ourselves to only one
- We need to be able to *identify* the component (so we can apply things like config and secrets, and possibly other modifications in the future), so basic targeting (id/name/ref) has been added.

Config and secrets application is not in this PR, but the structure is there and it should be in soon.

File: crates/wash-cli/src/dev.rs [hunk 1]
```diff
@@ -1639,58 +1631,313 @@ async fn run_dev_loop(
...
             format!("failed to generate a WADM manifest from (session [{session_id}])")
-        })?;
+        })?
+        .into_iter()
+        .collect::<Vec<_>>();
+
+    // Write out manifests to local files if a manifest output dir was specified
+    if let Some(output_dir) = &manifest_output_dir {
+        for manifest in manifests.iter() {
+            ensure!(
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs:1656]: Hmnn maybe -- I don't generally expect tools to make folders for me though if I specify one, normally they tell me the folder is missing... For example someone could try and give a path that is actually the path to a file, we'd want to fail in that case (not try to create the directory)

File: crates/wash-cli/src/dev.rs [hunk 2]
```diff
@@ -1639,58 +1631,313 @@ async fn run_dev_loop(
...
+            // Add any named secrets that are not present
+            if !secrets.iter().any(|c| c.name == *name) {
+                secrets.push(SecretProperty {
+                    name: name.to_string(),
+                    properties: source.clone(),
+                })
+            }
+        }
+        DevSecretSpec::Values { name, values } => {
+            // TODO: Create the secret in the store
```
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs]: Captured them all in the #2508 issue!

File: crates/wash-cli/src/dev.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (vados-cosmonic) [reply crates/wash-cli/src/dev.rs]: I'm definitely open to following up on this if it's a common stumbling block for people.

**Review by thomastaylor312 — APPROVED** [review]

**Pull request merged**